### DISCUSSION TITLE
How Gutenberg works for rendering

### DISCUSSION THREAD

**Discussion by scaccogatto** [WordPress/gutenberg#46234] [Q&A]
Hello,

I am studying this piece of software and I am wondering if there is any explanation on how Gutenberg renders the JS into HTML and saves it into the DB.

I am particularly interested in the architecture flow and the point where the magic happens.

Do you have anything to begin with?

**Comment by devfle** [comment]
Hey, maybe this documentation will be helpful to you: https://developer.wordpress.org/block-editor/explanations/architecture/data-flow/
